### ISSUE TITLE
Feature: Support separate changelog per project in monorepo

### ISSUE THREAD

**Issue by lasse-seivaag** [versionize/versionize#137]
# User stories

## Reader of changelogs
As a reader of changelogs
I want changelogs to only contain changes for the project I am viewing, when the project is part of a monorepo
Such that the changelog accurately reflects the changes made in the project I am currently viewing, without changes from other projects

## Developer in a monorepo
As a developer in a monorepo
I want options to configure which commit messages are included in the changelog, based on the path of the changed file.
Such that the changelog only includes changes made to the project(s) at the configured paths.

# Background 
We want to use versionize to automate versioning and changelog generation in our projects. We have a repository with several projects, with separate CI/CD pipelines.

From what I could find in the documentation, and skimming through the source code, am I correct that versionize does not currently filter changes based on paths of the changed files?

**Comment by Barsan1** [comment]
Hi, versionize is support mono repo.  
I will give an example of how you can achieve this be noted that it still have some pros and cons.  

## Steps:
1. initialize folder structure with git (powershell) and .versionize config file 
```powershell
# you can make it smaller but i don't want to skip any step
mkdir version && cd version && git init && dotnet new gitignore && mkdir src && cd src && dotnet new console --name Console && dotnet new classlib --name Library
```
structure should now look like this:  
```
| .gitignore
| .versionize
\---src
    \---Console
    |   |   Console.csproj
    |   |   Program.cs
    \---Library
        |   Class1.cs
        |   Library.csproj
```
2. add Version tag to projects (csproj)
```csharp
// Console && Library files gives version 0.0.0
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <Version>0.0.0</Version>
  </PropertyGroup>

</Project>
```
3. edit .versionize config file
```json
{
    "changelogAll": true,
    "skipDirty": true,
    "aggregatePreReleases": true,
    "silent": false,
    "projects": [
        {
            "name": "library",
            "path": "src/Library",
            "tagTemplate": "{name}-v{version}",
            "changelog": {
                "header": "Library Changelog",
                "sections": [
                    {
                        "type": "feat",
                        "section": "✨ Features",
                        "hidden": false
                    },
                    {
                        "type": "fix",
                        "section": "🐛 Bug Fixes",
                        "hidden": false
                    },
                    {
                        "type": "perf",
                        "section": "🚀 Performance",
                        "hidden": false
                    }
                ]
            }
        },
        {
            "name": "console",
            "path": "src/Console",
            "tagTemplate": "{name}-v{version}",
            "changelog": {
                "header": "Console Changelog",
                "sections": [
                    {
                        "type": "feat",
                        "section": "✨ Features",
                        "hidden": false
                    },
                    {
                        "type": "fix",
                        "section": "🐛 Bug Fixes",
                        "hidden": false
                    },
                    {
                        "type": "perf",
                        "section": "🚀 Performance",
                        "hidden": false
                    }
                ]
            }
        }
    ]
}
```
We have two projects: a console application and a class library. Both projects are located in their respective folders, each pointing to its own .csproj file. Additionally, take a look at the project template, which you can modify here if needed. The project names are also important, as they will help the versionize tool generate tags and changelogs automatically


4. work with git cli and versionize tool
```powershell
# add console project and commit
git add src/Console
git commit -m "feat: add console project"
# add .gitignore file and commit
git add .gitignore
git commit -m "chore: add gitignore file"
# add .versionize file and commit
git add .versionize
git commit -m "chore: add .versionze config file"
# its time to make version note that only first commit will be included in the changelog because its in the folder in src/Console and csproj file is there
versionize --proj-name "console"
# the command will give a tag of console-v0.0.0 and create changelog with only the feat commit (other commits not included)

# add library project and commit
git add src/Library
git commit -m "feat: add library project"

# show commits log until now
git log --oneline 

# 518daa9 feat: add library project
# db693a2 (tag: console-v0.0.0) chore(release): 0.0.0
# b5fc6e4 chore: add .versionize config file
# 5666e9b chore: add .gitignore file
# 601aa70 feat: add console library

# Note the chore(release) tag made by the versionize tool
```
5. fixing some issues with the projects (console & library)


in `Class1.cs`
```csharp
namespace Library;

public class Class1
{
    // add this 
    public void Fix1()
    {
        
        System.Console.WriteLine("add fix for console");
    }
}
```
in `Program.cs`
```csharp
// See https://aka.ms/new-console-template for more information
Console.WriteLine("Hello, World!");
Console.WriteLine("Fixing this"); // add this
```
6. time to commit and release
```powershell
# add modified Program.cs file
git add src/Console
git commit -m "fix: new fix in console"
# add modified Class1.cs
git add src/Library
git commit -m "fix: new fix in library"
# make new version for console project now the version will be 0.0.1 and update the changelog file
versionize --proj-name "console"

# show updated commit log
git log --oneline

# e3fafbf (tag: console-v0.0.1) chore(release): 0.0.1
# d628fac fix: new fix in library
# 8d3f0d3 fix: new fix in console
# 518daa9 feat: add library project
# db693a2 (tag: console-v0.0.0) chore(release): 0.0.0
# b5fc6e4 chore: add .versionize config file
# 5666e9b chore: add .gitignore file
# 601aa70 feat: add console library

# Note the new chore(release) tag made by the versionize tool

# its time to release the library project it will be first version 0.0.0 and create new changelog file in the Library folder.
versionize --proj-name "library"

# Note this will add the feat commit (518daa9) and also the fix commit (d628fac) to the changelog

# show updated commit log
git log --oneline

# 82ec57f (HEAD -> main, tag: library-v0.0.0) chore(release): 0.0.0
# e3fafbf (tag: console-v0.0.1) chore(release): 0.0.1
# d628fac fix: new fix in library
# 8d3f0d3 fix: new fix in console
# 518daa9 feat: add library project
# db693a2 (tag: console-v0.0.0) chore(release): 0.0.0
# b5fc6e4 chore: add .versionize config file
# 5666e9b chore: add .gitignore file
# 601aa70 feat: add console library
```

## Final
Now we have 2 changelog files one that track the Console project and one that track the Library project.  
This is the final structure
```
| .gitignore
| .versionize
\---src
    \---Console
    |   |   CHANGELOG.md
    |   |   Console.csproj
    |   |   Program.cs
    \---Library
        |   CHANGELOG.md
        |   Class1.cs
        |   Library.csproj
```

**Comment by lasse-seivaag** [comment]
@Barsan1 
Thank you so much for the extremely detailed example! :D

Looks like it worked wonders
